app/controllers/account_controller.rb

```diff
@@ -587,7 +587,7 @@ def api_keys
   end
 
   def create_api_key
-    @key = APIKey.new(params[:key].permit!)
+    @key = APIKey.new(params.require(:key).permit(:user_id, :notes))
```

Comment:
I don't think this solves the real problem that was causing the test to fail before.  What you've done with this change is change the error message from:
`#<NoMethodError: undefined method `permit!' for nil:NilClass

    @key = APIKey.new(params[:key].permit!)
                                  ^^^^^^^^>
`
to
`#<ActionController::ParameterMissing: param is missing or the value is empty: key
Did you mean?  action
               controller
               commit>
`

While this means that the e.to_str on line 598 of account_controller.rb no longer out right fails in test/test_helper.rb, the error string could still fail in the same.  The real issue is on 596.  The error getting raised is going through the translation process when it shouldn't be.  I think if we changed the definition of `account_api_keys_create_failed` in the locales/*.txt, to not have the `[msg]` bit and instead just appended the msg to the end it would be a more robust solution.  Essentially make `flash_error(:account_api_keys_create_failed.t + e.to_s)` work.  While you could argue that there might some obscure language where the error message belongs somewhere in the middle of the text, that is not the case for any of our existing translations and I think it would be reasonably understood regardless.

Reply:
See Nathan's commit which accomplishes this.